src/main/java/com/example/bookstore/service/shoppingcart/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,80 @@
+package com.example.bookstore.service.shoppingcart.impl;
+
+import com.example.bookstore.dto.cartitem.CartItemDto;
+import com.example.bookstore.dto.cartitem.CreateCartItemRequestDto;
+import com.example.bookstore.dto.cartitem.UpdateCartItemRequestDto;
+import com.example.bookstore.dto.shoppingcart.ShoppingCartDto;
+import com.example.bookstore.exception.EntityNotFoundException;
+import com.example.bookstore.mapper.CartItemMapper;
+import com.example.bookstore.mapper.ShoppingCartMapper;
+import com.example.bookstore.model.CartItem;
+import com.example.bookstore.model.ShoppingCart;
+import com.example.bookstore.model.User;
+import com.example.bookstore.repository.cartitem.CartItemRepository;
+import com.example.bookstore.repository.shoppingcart.ShoppingCartRepository;
+import com.example.bookstore.service.shoppingcart.ShoppingCartService;
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemRepository cartItemRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final CartItemMapper cartItemMapper;
+
+    @Override
+    public ShoppingCartDto getShoppingCart(Long id) {
+        ShoppingCart shoppingCart = getShoppingCartByUserId(id);
+        return shoppingCartMapper.toDto(shoppingCart);
+    }
+
+    @Override
+    @Transactional
+    public ShoppingCartDto addBookToShoppingCart(
+            Long id, CreateCartItemRequestDto cartItemRequestDto) {
+        ShoppingCart shoppingCart = getShoppingCartByUserId(id);
+        CartItem cartItem = cartItemMapper.toEntity(cartItemRequestDto);
+        cartItem.setShoppingCart(shoppingCart);
+        cartItemRepository.save(cartItem);
+        return shoppingCartMapper.toDto(shoppingCart);
+    }
+
+    @Override
+    @Transactional
+    public CartItemDto updateCartItem(
+            Long id, Long cartItemId, UpdateCartItemRequestDto cartItemRequestDto) {
+        CartItem cartItem = getCartItemById(cartItemId);
+        cartItem.setQuantity(cartItemRequestDto.getQuantity());
+        CartItem savedCartItem = cartItemRepository.save(cartItem);
+        return cartItemMapper.toDto(savedCartItem);
+    }
+
+    @Override
+    @Transactional
+    public void removeCartItem(Long cartItemId) {
+        CartItem cartItem = getCartItemById(cartItemId);
+        cartItemRepository.delete(cartItem);
```

Comment:
Be sure, that cartItem will be removed not only from DB but also from shop cart Set<CartItem> cartItems